### ISSUE TITLE
Artifacts when applying fine-tuned model to large image

### ISSUE THREAD

**Issue by maxeyje** [computational-cell-analytics/micro-sam#1114]
After fine-tuning the model with an image size of (4096, 6144), I am attempting to run it on a very large sample of size (80896,242688). In the automatic instance segmentation function, the tile_size is set to [3596, 5644] with a halo of [500, 500]. Unfortunately, this produces a large number of errors/artifacts, especially in regions that are all 0 (even though examples of edge images were included in the fine-tuning set). Do you have any recommendations to improve the application? Thank you for your help!

**Comment by maxeyje** [comment]
Thank you for the quick response! The data is 2D EM and we are segmenting the nucleus. The fine-tuning was completed on images that were (4096, 6144) without any tiling (so I am assuming they were internally downsampled to (1024, 1024) based on your response). Interestingly, the fine-tuned model performs almost perfectly on test images (4096, 6144). It is only when I used the tiling function on the larger image that the artifacts appeared. The training tiles were cropped regions from the larger image so the objects in the two data sets should be the same size. The nuclei are on the order of 1286x2361 pixels with significant variance.

**Comment by anwai98** [comment]
Hi @maxeyje,

Thanks for the detailed context. I understand the reason for choosing the tiling heuristics now.

> The training tiles were cropped regions from the larger image so the objects in the two data sets should be the same size.

That is indeed true for the trained model. It is hard to say though what artifacts are getting picked up.

Is the resolution for the trained images and the test images close enough?

Also, would it be okay for you if you could send me the trained model and one or two images? (I understand the images are too massive, a relatively smaller crop with a few nuclei in there should also do the trick). I just wanna see what are the artifacts being picked and if it's easier to get rid of them by changing some parameters! (if it's okay, you can share them to me via email: anwai.archit@gmail.com)

**Comment by maxeyje** [comment]
Thanks for your response. The resolution for the training and test sets are identical. The only difference is the overall size of the images. I will see if I can get permission to share images. I appreciate your help!

**Comment by anwai98** [comment]
> The only difference is the overall size of the images.

How different are we talking? (could you give me an estimate spatial shapes of training and test images)

> I will see if I can get permission to share images. I appreciate your help!

No worries if it's too much stress. It'll be slightly tricky to communicate taking guesses about the data and targets, but if you could find one of your publications / another publication in a similar direction, that should do the trick too!

Another thing you could try (assuming you are giving automatic segmentation a shot) is to observe what are the artifacts. Are they small dots / regions / surface areas which are quite far from being a nucleus? If these regions are small in general, you can increase the `min_size` argument (should be a part of the GUI / CLI / python scripts) to filter out these small regions. If it's picking up another structure, then the question comes how did the training go (i.e. how many training samples did the loader generate, what was the total number of epochs the model was trained for).

Hope this helps! (I'll leave this issue open for now in case you have any concrete questions)

**Comment by anwai98** [comment]
> The resulting objects look like nuclei in shape and size but are on part of the image with no data.

That's indeed concerning. It's hard to say why it pops up though (could be a reason for a severe downsizing of images / some strange border artifacts picked up by the model / normalization artifact). The easiest fix here would be to match the segmentations with the image data and check for whether the segmentation is in a valid intensity region, otherwise throw them out! (the simplest heuristic I can think of to filter these out)

**Comment by anwai98** [comment]
Hi @maxeyje,

Any updates on this? Since we had a good discussion about the details on your data, I'll assume that the problems are sorted out (?)

In case you have further things to discuss, please open a new issue and we'll be happy to help!

And thanks for your interest in `micro-sam` once again!
